src/dss_cell.hpp

```diff
@@ -0,0 +1,82 @@
+#pragma once
+#pragma once
+
+#include <vector>
+
+#include <common_types.hpp>
+#include <util/debug.hpp>
+
+namespace nest {
+namespace mc {
+
+/// data spike source: A cell that generates spikes provided as a vector of
+/// floating point valued spiketimes at the start of a run
+
+class dss_cell {
+public:
+    struct dss_cell_description {
```

Comment:
I don't know if they need to be in separate files for now, however they should be distinct classes. In practice, the description is used in the front end, and directly manipulated/built by the user or application driving the library.

Here the front end description is part of the back end implementation. For a `dss_cell` we will probably have just the one back end implementation, however for many cell types we could have multiple solvers and back end combinations, and you would be forced to separate the definitions. For consistency, they should always be separate.